fix: third party dependency upgrade delay

MatheusBaldi opened the pull request:
https://github.com/silvermine/renovate-config/issues/2

onebytegone commented on default.json:
```diff
@@ -10,7 +10,8 @@
          "matchPackageNames": [ ".*" ],
-         "minimumReleaseAge": "7 days"
+         "minimumReleaseAge": "7 days",
+         "internalChecksFilter": "strict"
```
Just checking, isn't [`strict` the default for `internalChecksFilter`](https://docs.renovatebot.com/configuration-options/#internalchecksfilter)? Additionally, it doesn't seem `internalChecksFilter` is supported on `packageRules` items ([docs](https://docs.renovatebot.com/configuration-options/#packagerules), [types](https://github.com/renovatebot/renovate/blob/9f600d14457b3d99b08b255420c56cfda7dd2bf2/lib/config/types.ts#L362)).

MatheusBaldi replied:
@onebytegone 
Regarding if the field is supported or not, my TS is a bit rusty, but I believe it allows any key-value pair because of the `extends... Record<string, unknown>` in [line 365](https://github.com/renovatebot/renovate/blob/9f600d14457b3d99b08b255420c56cfda7dd2bf2/lib/config/types.ts#L365). The same happens in the [RenovateConfig interface](https://github.com/renovatebot/renovate/blob/9f600d14457b3d99b08b255420c56cfda7dd2bf2/lib/config/types.ts#L224C8-L224C18). The usage of `internalChecksFilter` is covered by the [package tests](https://github.com/renovatebot/renovate/blob/9f600d14457b3d99b08b255420c56cfda7dd2bf2/lib/workers/repository/process/lookup/filter-checks.spec.ts#L182-L183), so it should work.
But I've now searched through some of their tests, and it seems it should also work without defining it, probably using `strict` as default value. However, I couldn't find where the default value is defined in their code.

I've searched some other options and there seems to be a problem with our regex for the matchPackageNames. It seems more likely to be the cause of the problem, so I'll stash these changes locally and commit a new one with the regex fix

onebytegone commented on renovate.json:
```diff
@@ -1,4 +0,0 @@
-{
```
Why is this being deleted?

MatheusBaldi replied:
@onebytegone 
[In their docs](https://docs.renovatebot.com/config-overview/#onboarding-config), when talking about config repositories for the bot, it's said that `renovate.json` has been deprecated, and for now it is only used when a `default.json` is not found. I couldn't test locally, but my assumption was that there could possibly be a bug in this fallback behavior. I've double checked their fallback implementation and it seems fine though, so this deletion was more of a test in this PR context.
Also, it is a cyclical reference to this same repository, which if used, should lead to `default.json` which is being kept, or, if somehow `default.json` is not used, it will fallback to itself (`renovate.json`) again and enter a loop.

But there is another possible fix I'd like to try. Should I revert this change for now?

MatheusBaldi commented on default.json:
```diff
@@ -9,12 +9,12 @@
       {
          "groupName": "silvermine packages",
-         "matchPackageNames": [ "^@silvermine/" ],
+         "matchPackageNames": [ "/@silvermine\//" ],
```
Besides adding the missing slashes, I removed the preceding carret because this pattern was not being captured in [regex101.com](https://regex101.com/?flavor=javascript&flags=ginst), which is the regex online runner recommended by them to test https://docs.renovatebot.com/string-pattern-matching/#use-regex101-to-test-your-patterns. I tested it using the package.json from [standardization](https://github.com/silvermine/standardization/blob/master/package.json) and [onebytegone/cloud-utils/package.json](https://github.com/onebytegone/cloud-utils/blob/master/package.json)

onebytegone replied:
```suggestion
         "matchPackageNames": [ "/^@silvermine\//" ],
```

onebytegone commented on default.json:
```diff
@@ -9,12 +9,12 @@
       {
          "groupName": "all dependencies",
-         "matchPackageNames": [ ".*" ],
+         "matchPackageNames": [ "/.*/" ],
```
```suggestion
         "matchPackageNames": [ "*" ],
```

MatheusBaldi committed:
fix: adjust matchPackageNames match strings
How matching works: https://docs.renovatebot.com/configuration-options/\#matchpackagenames
How to format regex: https://docs.renovatebot.com/string-pattern-matching/\#regex-matching
